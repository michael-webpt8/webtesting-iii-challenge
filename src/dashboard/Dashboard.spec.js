// Test away
import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Dashboard from './Dashboard';
import { cleanup } from 'react-testing-library';

afterEach(cleanup);

test('Renders display correctly', () => {
  expect(render(<Dashboard />).asFragment()).toMatchSnapshot();
});

test('renders closed on click of close gate', async () => {
  // open message first
  // click button fire async operation
  // click button to fire async operation async/await
  // assert closed message is now on the DOM
  const { getByText, getByTestId } = render(<Dashboard />);

  const closedMessage = getByTestId('gate-feature');
  expect(closedMessage).not.toBeNull();

  const button = getByTestId('gate-feature');
  await fireEvent.click(button);

  expect(getByText(/closed/i));
});

test('renders Locked when locking gate after closing gate', async () => {
  const { getByText, getByTestId } = render(<Dashboard />);
  const toggleGate = getByTestId('gate-feature');
  const lockGate = getByTestId('locking-feature');

  const toggleGateButton = toggleGate;
  const lockGateButton = lockGate;

  fireEvent.click(toggleGateButton);
  expect(getByText(/closed/i));
  await fireEvent.click(lockGateButton);
  expect(getByText(/locked/i));
});
