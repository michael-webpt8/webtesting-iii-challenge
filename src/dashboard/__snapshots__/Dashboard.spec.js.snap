// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders display correctly 1`] = `
<DocumentFragment>
  <div
    class="display panel"
  >
    <div
      class="led green-led"
    >
      Unlocked
    </div>
    <div
      class="led green-led"
    >
      Open
    </div>
  </div>
  <div
    class="controls panel"
  >
    <button
      class="toggle-btn"
      data-testid="locking-feature"
      disabled=""
    >
      Lock Gate
    </button>
    <button
      class="toggle-btn"
      data-testid="gate-feature"
    >
      Close Gate
    </button>
  </div>
</DocumentFragment>
`;
